### DISCUSSION TITLE
[RFM69/SX1278] Allow to set the RSSI Threshold for the RSSI interrupt

### DISCUSSION THREAD

**Discussion by obones** [jgromes/RadioLib#558] [Ideas]
Hello,

Following the merge of #542, one can now configure a DIO pin to output the RSSI interrupt which gets asserted when the received RSSI raw value is above a given threshold.
Unfortunately, there is currently no way to set that threshold value as there is no public method for it.
In the RFM69, the value is stored in the `RADIOLIB_RF69_REG_RSSI_THRESH` register and I can easily write a `setRSSIThreshold(byte value)` method.
But reading the datasheet and the existing `RF69:getRSSI()` method code, I see that RadioLib is making it easier for end users by dividing by two and negating the register value when retrieving the current RSSI value.
I'm thus wondering if the newly written `setRSSIThreshold` method should instead accept a `float` and multiply the received value by `-2.0` so that end users send a "user friendly" RSSI value while the method ensures a valid register value is set.

I checked and this division by -2 is also done for the SX1278 family and the SX1231 (derived from RFM69), so I'm suggesting to provide the same functionality on those chips as well.

What do you think?

**Comment by jgromes** [comment]
In general, I always try to stick as much as possible to physical units - IMHO, getRSSI should return RSSI in dBm, and setRSSIThreshold should take argument in the same unit. Sometimes arbitrary units are inevitable, but that's not the case here, so I would prefer a `setRSSIThreshold(float dbm)` method.

**Reply by obones** [reply]
Totally makes sense, just wanted to be sure before submitting a PR so as not to avoid wasting anyone's time

**Reply by obones** [reply]
Done: https://github.com/jgromes/RadioLib/pull/561
